src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/BuildPropertyPage/TrimmingEnumProvider.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.Build.Framework.XamlTypes;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties;
+
+[ExportDynamicEnumValuesProvider("TrimmingEnumProvider")]
+[AppliesTo(ProjectCapability.DotNet)]
+internal class TrimmingEnumProvider : IDynamicEnumValuesProvider
+{
+    public Task<IDynamicEnumValuesGenerator> GetProviderAsync(IList<NameValuePair>? options)
+    {
+        return Task.FromResult<IDynamicEnumValuesGenerator>(new TrimmingEnumGenerator());
+    }
+
+    private class TrimmingEnumGenerator : IDynamicEnumValuesGenerator
+    {
+        public bool AllowCustomValues => true;
+
+        public Task<ICollection<IEnumValue>> GetListedValuesAsync()
+        {
+            List<IEnumValue> enumValues = new()
+        {
```

Comment:
Indentation

Reply:
if it doesn't change store it in static field on class?

Reply:
oh, that's a good idea, adding it in next commit. 🙂

Reply:
This PR is good to go as-is, but if you wanted to make further optimisations here, you could actually have this field be `Task<ICollection<IEnumValue>>` so that `GetListedValues` doesn't have to allocate.

The same is true for `TrimmingEnumProvider.GetProviderAsync`.